cc-raisely-components/components/facil-conversation-list/facil-conversation-list.js

```diff
@@ -152,43 +235,148 @@
 
 		toggleFilter = () => {
 			// If we're about to switch to filtered
-			this.setState({	filter: !this.state.filter }, this.setConversations);
-		}
+			this.setState(
+				{ filter: !this.state.filter },
+				this.setConversations
+			);
+		};
+
+		toggleSelectConversations = () => {
+			//If we want to select duplicates to mergE
+			this.setState({
+				showCheckbox: !this.state.showCheckbox,
+			});
+		};
+
+		checkConversationAlreadyExists = (conversation) => {
+			const { selectedConversations } = this.state;
+
+			return selectedConversations.some(
+				(convo) => convo.uuid === conversation.uuid
+			);
+		};
+
+		setSelectedConversations = (conversation) => {
+			const { selectedConversations } = this.state;
+
+			if (this.checkConversationAlreadyExists(conversation)) {
+				const convo = selectedConversations.filter(
+					(convo) => convo.uuid !== conversation.uuid
+				);
+
+				this.setState({ selectedConversations: convo });
+			} else {
+				this.setState({
+					selectedConversations: [
+						...selectedConversations,
+						conversation,
+					],
+				});
+			}
+		};
+
+		mergeConversations = async () => {
+			const { selectedConversations } = this.state;
```

Comment:
With all the API calls being made in the cloud function, this will likely take several seconds. So give some thought to how you want to indicate to the user that a merge is in progress while you wait for the request to return.
That reminds me, we probably need to extend the default timeout on the cloud function (which I think is 30 seconds). If latency is bad some days, this could timeout